src/azul/service/elasticsearch_service.py

```diff
@@ -416,6 +416,7 @@ def apply_paging(self,
                            _generate_paging_dict can know if there is another
                            page. Use this to prevent a last page that's empty.
         """
+        require(isinstance(peek_ahead, bool), 'peek_ahead not a bool', peek_ahead)
```

Comment:
I think an `assert` would be more appropriate here than a `require`. The value of `peek_ahead` is only set from within our own code so there's no But IIRC Hannes was present when you worked out this line so I won't request you to change it. Just something to think about next time Hannes looks at it.

Reply:
Correct. A failing assertion would indicate a bug, not a user error.

```suggestion
        assert isinstance(peek_ahead, bool), type(peek_ahead)
```